packages/cli/src/controller/generate-controller.ts

```diff
@@ -152,6 +160,23 @@ export function constructDatasources(userInput: UserInput): EthereumDs {
 
   const assets = new Map([[abiName, {file: userInput.abiPath}]]);
 
+  if (isTs) {
```

Comment:
I think for readability, reduced union types and easier testing it would be worth splitting these functions into yaml and TS versions